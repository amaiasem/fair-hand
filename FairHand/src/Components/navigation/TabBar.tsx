import React, { useState } from 'react'
import { View, StyleSheet } from 'react-native'
import Tab from './Tab'
import { COLOR, SHADOW } from '../../../constants'

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    bottom: 0,
    width: '100%',
    backgroundColor: COLOR.white,
    ...SHADOW
  },
  container: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: 50
  }

})

const TabBar = ({ state, navigation }: any) => {
  const [selected, setSelected] = useState('Home')
  const { routes } = state

  const handlePress = (activeTab: string, index: number) => {
    if (state.index !== index) {
      setSelected(activeTab)
      navigation.navigate(activeTab)
    }
  }

  return (
    <View style={styles.wrapper}>
      <View style={styles.container}>
        {
          routes.map((route: any, index: number) => (
            <Tab
              icon={route.params.icon}
              onPress={() => handlePress(route.name, index)}
              key={route.key}/>
          ))
        }
      </View>
    </View>
  )
}

export default TabBar
