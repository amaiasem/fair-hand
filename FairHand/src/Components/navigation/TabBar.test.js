import React from 'react'
import { cleanup, fireEvent, render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'
import TabBar from './TabBar.tsx'

describe('Given a component TabBar', () => {
  let component
  let navigate

  const routes = [{ key: 1, name: 'Home', params: { icon: 'home' } },
    { key: 2, name: 'Maps', params: { icon: 'map' } },
    { key: 3, name: 'User', params: { icon: 'user' } }]

  beforeEach(() => {
    navigate = jest.fn()
    component = (
        <NavigationContainer>
            <TabBar state={ { routes } } navigation={{ navigate }}/>
        </NavigationContainer>
    )
  })

  afterEach(() => cleanup())
  it('It should match the snapshot', () => {
    render(component)
    expect(component).toMatchSnapshot()
  })

  describe('When pressing the tab Home', () => {
    it('It should call handlePress', () => {
      const { getAllByTestId } = render(component)
      fireEvent.press(getAllByTestId('button-tab')[0])
      expect(navigate).toHaveBeenCalledWith('Home')
    })
  })
})
