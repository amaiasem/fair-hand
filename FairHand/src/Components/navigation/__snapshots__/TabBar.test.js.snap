// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a component TabBar It should match the snapshot 1`] = `
<ForwardRef(NavigationContainer)>
  <TabBar
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    state={
      Object {
        "index": 1,
        "routes": Array [
          Object {
            "name": "Home",
            "params": Object {
              "icon": "home",
            },
          },
          Object {
            "name": "Maps",
            "params": Object {
              "icon": "map",
            },
          },
          Object {
            "name": "User",
            "params": Object {
              "icon": "user",
            },
          },
        ],
      }
    }
  />
</ForwardRef(NavigationContainer)>
`;
